lib/Doctrine/ORM/Tools/SchemaValidator.php

```diff
@@ -406,6 +407,28 @@ function (array $fieldMapping) use ($class): ?string {
                             );
                         }
 
+                        if (
+                            isset($fieldMapping['enumType'])
+                            && $propertyType !== $fieldMapping['enumType']
+                            && interface_exists($propertyType)
```

Comment:
Why do we need to check that it's an interface?

Reply:
For two reasons: 1) The static analyzer complained about passing a non-class-string as the second argument of the `is_a` function; 2) It's only possible for enums to implement interfaces.